elements/radio-group/radio-group.js

```diff
@@ -1,151 +1,233 @@
+     * already set to `tabindex=0`. This makes the first `RadioButton` focusable.
+     * This assumes `RadioButton` children are already in the DOM and their
+     * definition has already been loaded. For a more robust implementation you
+     * might consider using a Mutation Observer to detect if children are
+     * present yet. The `RadioGroup also adds listeners for keyboard and mouse
+     * events. 
+     */
+    connectedCallback() {
+      this.setAttribute('role', 'radiogroup');
+      let hasCheckedChild = this.querySelector('[role="radio"][tabindex="0"]');
+      if (!hasCheckedChild) {
+        this.querySelector('[role="radio"]').setAttribute('tabindex', 0); 
+      }
+      this.addEventListener('keydown', this._onKeyDown);
+      this.addEventListener('click', this._onClick);
+    }
 
-  }
+    /**
+     * If the `RadioGroup` is removed from the DOM, clean up any event
+     * listeners.
+     */
+    disconnectedCallback() {
+      this.removeEventListener('keydown', this._onKeyDown);
+      this.removeEventListener('click', this._onClick);
+    }
 
-  onKeyDown(e) {
-    switch (e.keyCode) {
-      case KEYCODE.UP:
-      case KEYCODE.LEFT:
-        {
-          // We will be doing all the handling, so we don’t want
-          // the browser to do anything for us.
+    /**
+     * If the user pressed an arrow key, call preventDefault to prevent the
+     * page from scrolling. If the up or left arrow keys were pressed, select
+     * the previous `RadioButton`. If the down or right keys were pressed,
+     * select the next `RadioButton`.
+     */
+    _onKeyDown(e) {
+      switch (e.keyCode) {
+        case KEYCODE.UP:
+        case KEYCODE.LEFT:
           e.preventDefault();
-          _this.setCheckedToPrevButton();
+          this._setCheckedToPrevButton();
           break;
 
-        }
-
-      case KEYCODE.DOWN:
-      case KEYCODE.RIGHT:
-        {
-          // Same as above.
+        case KEYCODE.DOWN:
+        case KEYCODE.RIGHT:
           e.preventDefault();
           this._setCheckedToNextButton();
           break;
-        }
-
+        
+        default:
+          break;
+      }
     }
-  }
-
-  get selectedRadioButton() {
-    return this.querySelector('[aria-checked="true"]') || this.firstRadioButton;
-  }
 
-  get lastRadioButton() {
-    return Array.from(this.querySelectorAll('[role="radio"]')).pop();
-  }
+    /**
+     * A getter for whichever `RadioButton` is currently checked. If no button
+     * is checked, return the first `RadioButton` child.
+     */
+    get checkedRadioButton() {
+      return this.querySelector('[aria-checked="true"]') || this.firstRadioButton;
+    }
 
-  get firstRadioButton() {
-    return Array.from(this.querySelectorAll('[role="radio"]')).shift();
-  }
+    /**
+     * A getter for the first `RadioButton` child.
+     */
+    get firstRadioButton() {
+      return Array.from(this.querySelectorAll('[role="radio"]')).shift();
```

Comment:
_Arguably_ a little simpler (and memory-conservative?) to use `[role="radio"]:first-of-type`. WDYT?